docs/src/pages/flags.md

```diff
@@ -98,6 +108,10 @@ const linter = new Linter({
 });
 ```
 
+::: tip
+The `ESLint` class also reads the `ESLINT_FLAGS` environment variable to set flags.
+:::
```

Comment:
If a tool needs to create an `ESLint` instance without a flag that is set in the environment variable, what will it do? I believe the only way to instantiate the `ESLint` class with a predictable set of flags will be modifying the current process environment.

```js
delete process.env.ESLINT_FLAGS;
const eslint = new ESLint({ flags: myFlagsNotTheirs });
```

Are we fine with this usage?

Reply:
I think that's okay as it seems like an edge case to me.

Reply:
Okay, then let's use the merging approach for the constructor option and the environment variable.